traveler-swift-core/TravelerKit/Models/CatalogSearchQuery.swift

```diff
@@ -0,0 +1,128 @@
+//
+//  CatalogSearchQuery.swift
+//  TravelerKit
+//
+//  Created by Omar Padierna on 2019-08-07.
+//  Copyright © 2019 Ata Namvari. All rights reserved.
+//
+
+import Foundation
+
+/// A general all encompassing search query
+public protocol SearchQuery {
+    /// Text for text-based queries
+    var text: String? { get }
+    /// Minimum allowed price for products
+    var minPrice: Double? { get }
+    /// Maximum allowed price for products
+    var maxPrice: Double? { get }
+    /// Product category
+    var categories: [Category]? { get }
+    /// Latitude belonging to the top left corner of bounding box
+    var topLeftLatitude: Double? { get }
```

Comment:
Can we put the latitudes and longtitudes in the GeoSearch protocol?

Reply:
Good one, I agree.